### PR TITLE
feat(appset): Add argocd.argoproj.io/application-set-name to rendered appset applications

### LINKED ISSUE #13309: argo cli should be able to show all apps of an appset
# Summary

argo cli should be able to show all apps of an appset

# Motivation

After changing an appset, I want to use the argocd cli to verify all apps of that appset are synced.

Currently, you can do this using (replacing `appset-name` with the name of your appset):

```
argocd app list -o json | jq --raw-output '.[] | select(.metadata.ownerReferences[0].name=="appset-name") | .metadata.name'
```

A simpler way would be nice.

# Proposal

Either:
* `argocd appset get <name>` should show all Applications owned by it
* `argocd app list` should accept a flag `--appset <appset-name>` to filter the output

### PR DESCRIPTION
**Pull request by MeNsaaH**
Note on DCO:

If the DCO action in the integration test fails, one or more of your commits are not signed off. Please click on the *Details* link next to the DCO action for instructions on how to resolve this.

Checklist:

* [ ] Either (a) I've created an [enhancement proposal](https://github.com/argoproj/argo-cd/issues/new/choose) and discussed it with the community, (b) this is a bug fix, or (c) this does not need to be in the release notes.
* [x] The title of the PR states what changed and the related issues number (used for the release note).
* [x] The title of the PR conforms to the [Toolchain Guide](https://argo-cd.readthedocs.io/en/latest/developer-guide/toolchain-guide/#title-of-the-pr)
* [x] I've included "Closes [ISSUE #]" or "Fixes [ISSUE #]" in the description to automatically close the associated issue.
* [ ] I've updated both the CLI and UI to expose my feature, or I plan to submit a second PR with them.
* [ ] Does this PR require documentation updates?
* [ ] I've updated documentation as required by this PR.
* [ ] Optional. My organization is added to USERS.md.
* [x] I have signed off all my commits as required by [DCO](https://github.com/argoproj/argoproj/blob/master/community/CONTRIBUTING.md#legal)
* [x] I have written unit and/or e2e tests for my change. PRs without these are unlikely to be merged.
* [ ] My build is green ([troubleshooting builds](https://argo-cd.readthedocs.io/en/latest/developer-guide/ci/)). 
* [ ] My new feature complies with the [feature status](https://github.com/argoproj/argoproj/blob/master/community/feature-status.md) guidelines.
* [x] I have added a brief description of why this PR is necessary and/or what this PR solves.

Please see [Contribution FAQs](https://argo-cd.readthedocs.io/en/latest/developer-guide/faq/) if you have questions about your pull-request.

As part of #13309, this PR adds a new label to all applications generated by appsets. This way, we can list all the apps belonging to an appset by running
```  
argocd app list -l argocd.argoproj.io/appset-instance=app-set-name
```

closes #13309

### CODE REVIEW

**Review by blakepettersson — APPROVED** [review]
This looks reasonable for me at least, at `$dayjob` we currently work around this by setting our own custom label. Let's see if my sentiment is shared by the others 😄

File: applicationset/controllers/applicationset_controller.go [hunk 1]
```diff
@@ -57,6 +57,10 @@ const (
 	//   https://github.com/argoproj-labs/argocd-notifications/blob/33d345fa838829bb50fca5c08523aba380d2c12b/pkg/controller/state.go#L17
 	NotifiedAnnotationKey             = "notified.notifications.argoproj.io"
 	ReconcileRequeueOnValidationError = time.Minute * 3
+
+	// LabelKeyAppSetInstance is the label key to use to uniquely identify the apps of an applicationset
+	// The ArgoCD applicationset name is used as the instance name
+	LabelKeyAppSetInstance = "argocd.argoproj.io/appset-instance"
```
Comment (crenshaw-dev) [inline applicationset/controllers/applicationset_controller.go:63]: I know it's more verbose, but for consistency with the existing `argocd.argoproj.io/application-set-refresh`, could we go with `argocd.argoproj.io/application-set-name`?

**Review by crenshaw-dev — CHANGES_REQUESTED** [review]
Thanks, @MeNsaaH!

File: applicationset/controllers/applicationset_controller.go (hunk 1 above)
Comment (MeNsaaH) [reply applicationset/controllers/applicationset_controller.go]: done @crenshaw-dev

**Review by crenshaw-dev — APPROVED** [review]

**Comment by dyatlov** [comment]
This PR introduced a regression: https://github.com/argoproj/argo-cd/issues/15282
It is no longer possible to have ApplicationSets with names longer than 63 chars.

I don't think forcing a custom label for an application was a good idea in the first place, honestly:
 1. because of the issue above
 2. because ApplicationSet is not creating applications synchronously and `argocd app list -l argocd.argoproj.io/appset-instance=app-set-name` (or its API analogue) can provide an incomplete view (especially if part of the applications are created and another part are not - which is the case for the issue above)

A better approach is to add wanna-be generated application names into ApplicationSet resource itself.
A *much* better approach is to also include application sync statuses next to application names within a given ApplicationSet.

**Pull request merged**